types/role.ts

```diff
@@ -13,61 +34,7 @@ export interface DefaultVotePolicy {
   weight?: string;
 }
 
-export type DaoPermission =
-  | '*:*'
-  | '*:Finalize'
-  | '*:AddProposal'
-  | '*:VoteApprove'
-  | '*:VoteReject'
-  | '*:VoteRemove'
-  | 'config:AddProposal'
-  | 'policy:AddProposal'
-  | 'add_bounty:AddProposal'
-  | 'bounty_done:AddProposal'
-  | 'transfer:AddProposal'
-  | 'call:AddProposal'
-  | 'vote:AddProposal'
-  | 'remove_member_from_role:AddProposal'
-  | 'add_member_to_role:AddProposal'
-  | 'upgrade_self:AddProposal'
-  | 'upgrade_remote:AddProposal'
-  | 'set_vote_token:AddProposal'
-  | 'config:VoteApprove'
-  | 'call:VoteApprove'
-  | 'upgrade_self:VoteApprove'
-  | 'upgrade_remote:VoteApprove'
-  | 'set_vote_token:VoteApprove'
-  | 'bounty_done:VoteApprove'
-  | 'policy:VoteApprove'
-  | 'add_bounty:VoteApprove'
-  | 'transfer:VoteApprove'
-  | 'vote:VoteApprove'
-  | 'remove_member_from_role:VoteApprove'
-  | 'add_member_to_role:VoteApprove'
-  | 'config:VoteReject'
-  | 'call:VoteReject'
-  | 'upgrade_self:VoteReject'
-  | 'upgrade_remote:VoteReject'
-  | 'set_vote_token:VoteReject'
-  | 'bounty_done:VoteReject'
-  | 'policy:VoteReject'
-  | 'add_bounty:VoteReject'
-  | 'transfer:VoteReject'
-  | 'vote:VoteReject'
-  | 'remove_member_from_role:VoteReject'
-  | 'add_member_to_role:VoteReject'
-  | 'config:VoteRemove'
-  | 'call:VoteRemove'
-  | 'upgrade_self:VoteRemove'
-  | 'upgrade_remote:VoteRemove'
-  | 'set_vote_token:VoteRemove'
-  | 'bounty_done:VoteRemove'
-  | 'policy:VoteRemove'
-  | 'add_bounty:VoteRemove'
-  | 'transfer:VoteRemove'
-  | 'vote:VoteRemove'
-  | 'remove_member_from_role:VoteRemove'
-  | 'add_member_to_role:VoteRemove';
+export type DaoPermission = `${'*' | ProposalKind}:${'*' | ProposalAction}`;
```

Comment:
Btw, TypeScript is complaining about this line, even though It should be a valid [template literal type](https://www.typescriptlang.org/docs/handbook/2/template-literal-types.html)

Reply:
I think this is because ProposalKind is not just a string. It is defined as  

export type ProposalKind =
  | {
      type: ProposalType.AddMemberToRole;
      memberId: string;
      role: string;
    }
  | {
      type: ProposalType.RemoveMemberFromRole;
      memberId: string;
      role: string;
    }
....

And in this case it is definetely cannot be part of a string

Reply:
Fixed by updating Prettier to latest.  But thanks for pointing out `ProposalKind`, didn't know there was already a type with that name

Reply:
<img width="1097" alt="image" src="https://user-images.githubusercontent.com/40799684/185402471-ae496251-2103-4dc8-996b-c59d0104f24e.png">
it still fails to compile due to prettier warnings. Is it a new rules?